
This fixes building 304.108 legacy diviers on 5.17+ kernels.

Patch developed(c) by Piotr Oniszczuk <piotr.oniszczuk@gmail.com> for
https://github.com/warpme/minimyth2

diff --speed-large-files --no-dereference --minimal -Naur NVIDIA-Linux-x86_64-340.108/kernel/conftest.sh NVIDIA-Linux-x86_64-340.108/kernel/conftest.sh
--- NVIDIA-Linux-x86_64-340.108/kernel/conftest.sh	2022-02-15 09:57:50.223285183 +0100
+++ NVIDIA-Linux-x86_64-340.108/kernel/conftest.sh	2022-02-15 09:46:22.109964261 +0100
@@ -1619,7 +1619,11 @@
             CODE="
             #include <linux/proc_fs.h>
             void conftest_PDE_DATA(void) {
+            #if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 17, 0)
+                pde_data();
+            #else
                 PDE_DATA();
+            #endif
             }"
 
             compile_check_conftest "$CODE" "NV_PDE_DATA_PRESENT" "" "functions"
diff --speed-large-files --no-dereference --minimal -Naur NVIDIA-Linux-x86_64-340.108/kernel/nv-linux.h NVIDIA-Linux-x86_64-340.108/kernel/nv-linux.h
--- NVIDIA-Linux-x86_64-340.108/kernel/nv-linux.h	2022-02-15 09:57:50.226618515 +0100
+++ NVIDIA-Linux-x86_64-340.108/kernel/nv-linux.h	2022-02-15 09:51:06.976625786 +0100
@@ -266,9 +266,9 @@
 #endif
 
 #ifdef CONFIG_PROC_FS
-#include <linux/proc_fs.h>
 #include <linux/seq_file.h>
 #endif
+#include <linux/proc_fs.h>
 
 #ifdef CONFIG_KDB
 #include <linux/kdb.h>
@@ -2052,8 +2052,12 @@
     })
 
 #if defined(NV_PDE_DATA_PRESENT)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 17, 0)
 # define NV_PDE_DATA(inode) PDE_DATA(inode)
 #else
+# define NV_PDE_DATA(inode) pde_data(inode)
+#endif
+#else
 # define NV_PDE_DATA(inode) PDE(inode)->data
 #endif
 
diff --speed-large-files --no-dereference --minimal -Naur NVIDIA-Linux-x86_64-340.108/kernel/uvm/conftest.sh NVIDIA-Linux-x86_64-340.108/kernel/uvm/conftest.sh
--- NVIDIA-Linux-x86_64-340.108/kernel/uvm/conftest.sh	2019-12-11 23:04:24.000000000 +0100
+++ NVIDIA-Linux-x86_64-340.108/kernel/uvm/conftest.sh	2022-02-15 09:55:01.226621573 +0100
@@ -1574,7 +1574,11 @@
             CODE="
             #include <linux/proc_fs.h>
             void conftest_PDE_DATA(void) {
+            #if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 17, 0)
+                pde_data();
+            #else
                 PDE_DATA();
+            #endif
             }"
 
             compile_check_conftest "$CODE" "NV_PDE_DATA_PRESENT" "" "functions"
